ocaml/testsuite/tests/typing-layouts-err-msg/error_messages.ml

```diff
@@ -76,28 +76,26 @@ Line 3, characters 21-22:
 Error: This expression has type v but an expression was expected of type
          'a t = ('a : void)
        The layout of v is immediate, because
-         it's an enumeration variant (all constructors are constant).
+         of the definition of v at line 2, characters 0-10.
```

Comment:
This is not a particularly helpful message, precisely because the type about which we are reporting is just a plain old `Tconstr`. Is there a way to detect such cases and avoid printing the history here? Doesn't have to be in this PR, but perhaps note this down as a future possible improvement.

Reply:
Opened a jira ticket to record this